### PR TITLE
peer2peer - General cleanup and updates

### PR DESCRIPTION
**Pull request by KaptenJansson**
**Description**
* Removed audio constraints for screen capture which caused Desktop and window capture not working.
* Replaced screen capture alert() with a <div> that has clickable links.
* Now removes video events before stopping video tracks.
* Moved the screen capture extension source to src/ folder and included a downloadable zip file.
* Updated the constraints format.
* Replaced the deprecated mediaStream.stop() with mediaStreamTrack.stop() handler.

**Purpose**
Maintenance and bug fixes.

### CODE REVIEW

**Comment by chenqiang1986** [comment]
Just see this pull request here.
For audio share with desktop capture, we've figured out the reason why it fails.
We have already landed chromium change.
And I've made https://github.com/webrtc/testrtc/pull/223 to fix it.

File: src/manual/peer2peer/index.html [hunk 1]
```diff
@@ -63,15 +63,24 @@
...
         </div>
         <div id="audio-source">
-          <label>Audio source <select class="drop-down" id="audiosrc"
-              onchange="updateGetUserMediaConstraints();"></select></label>
+          <div>
+            <label>Audio source</label>
+          </div>
+          <div>
+            <select class="drop-down" id="audiosrc"
+              onchange="updateGetUserMediaConstraints();"></select>
```
Comment (hkjellander) [inline src/manual/peer2peer/index.html]: +2 indent

File: src/manual/peer2peer/index.html [hunk 2]
```diff
@@ -63,15 +63,24 @@
...
         </div>
         <div id="video-source">
-          <label>Video source <select class="drop-down" id="videosrc"
-              onchange="updateGetUserMediaConstraints();"></select></label>
+          <div>
+            <label>Video source</label>
+          </div>
+          <div>
+            <select class="drop-down" id="videosrc"
+              onchange="updateGetUserMediaConstraints();"></select>
```
Comment (hkjellander) [inline src/manual/peer2peer/index.html]: +2 indent

File: src/manual/peer2peer/js/main.js [hunk 3]
```diff
@@ -249,13 +247,13 @@ function updateGetUserMediaConstraints() {
...
       if (devices.audioId !== null) {
-        global.constraints.audio = {optional: [{deviceId: devices.audioId}]};
+        global.constraints.audio = {deviceId: devices.audioId};
       }
     }
 
     if ($('video').checked) {
       if (devices.videoId !== null) {
-        global.constraints.video.optional.push({deviceId: devices.videoId});
+        global.constraints.video.deviceId = devices.videoId;
```
Comment (hkjellander) [inline src/manual/peer2peer/js/main.js]: Why is the style different than constraints.audio? Aren't they similar?

File: src/manual/peer2peer/js/main.js [hunk 4]
```diff
@@ -285,11 +283,14 @@ function clearLog() {
 // Stops the local stream.
 function stopLocalStream() {
   if (typeof global.localStream === 'undefined') {
-    error_('Tried to stop local stream, ' +
+    warning_('Tried to stop local stream, ' +
```
Comment (hkjellander) [inline src/manual/peer2peer/js/main.js]: +1 space indent on next line

File: src/manual/peer2peer/js/main.js [hunk 5]
```diff
@@ -378,24 +379,45 @@ function getDevices() {
...
-                        '(Download from here: http://goo.gl/kLEycY)\n' +
-                        '5. Reload this page over https';
-          alert(message);
-        }
-        window.postMessage({type: 'SS_UI_REQUEST', text: 'start'}, '*');
-      });
+  $('start-screencapture').addEventListener('click', function() {
+    // send screen-sharer request to content-script
+    window.postMessage({type: 'SS_UI_REQUEST', text: 'start'}, '*');
+    if (!extensionInstalled) {
```
Comment (hkjellander) [inline src/manual/peer2peer/js/main.js]: extensionInstalled is set to false above, how can it ever be true?

File: src/manual/peer2peer/js/main.js [hunk 6]
```diff
@@ -860,28 +873,40 @@ function doGetUserMedia_(constraints) {
...
-                 'fired.');
-      };
+        displayVideoSize(videoElement);}, true);
     }
   }).catch(function(error) {
     error_('GetUserMedia failed with error: ' + error.name);
   });
 }
 
+function registerVideoTrackEvents(stream) {
```
Comment (hkjellander) [inline src/manual/peer2peer/js/main.js]: What about the case when stream.getVideoTracks() contains more than one track? Maybe display a warning or even just wire up the same events for all video tracks?

Maybe stream -> track makes sense too.

File: src/manual/peer2peer/js/main.js [hunk 7]
```diff
@@ -1146,17 +1171,21 @@ function getEvaluatedJavaScript_(stringRepresentation) {
 function forceIsac_() {
   setOutgoingSdpTransform(function(sdp) {
     // Remove all other codecs (not the video codecs though).
-    sdp = sdp.replace(/m=audio (\d+) RTP\/SAVPF.*\r\n/g,
-                      'm=audio $1 RTP/SAVPF 104\r\n');
+    sdp = sdp.replace(/m=audio (\d+) UDP\/TLS\/RTP\/SAVPF.*\r\n/g,
+                      'm=audio $1 UDP\/TLS\/RTP\/SAVPF 104\r\n');
+    sdp = sdp.replace('a=rtcp-fb:111 transport-cc',
+                        'a=rtcp-fb:104 transport-cc');
```
Comment (hkjellander) [inline src/manual/peer2peer/js/main.js]: -2 spaces indent

File: src/manual/peer2peer/js/main.js (hunk 3 above)
Comment (KaptenJansson) [inline src/manual/peer2peer/js/main.js]: I did not consider the code around it, I've now made the video constraints format the same as audio.

File: src/manual/peer2peer/js/main.js (hunk 4 above)
Comment (KaptenJansson) [inline src/manual/peer2peer/js/main.js]: Done

File: src/manual/peer2peer/js/main.js (hunk 5 above)
Comment (KaptenJansson) [inline src/manual/peer2peer/js/main.js]: It's part of an event listener, it's set to true when a ping response from the extension is received https://github.com/webrtc/testrtc/pull/222/files/70b11656f4a3fc14dc977a69193bd86d11f4c06c#diff-39bad8872c33a8321fc56bcb9eee5587R430

File: src/manual/peer2peer/js/main.js (hunk 6 above)
Comment (KaptenJansson) [inline src/manual/peer2peer/js/main.js]: The whole page is constructed on this assumption and there is no functionality in the page to actually request a 2nd track (you would need to clone a track and add it to the mediastream)

**Comment by KaptenJansson** [comment]
@hkjellander PTAL

File: src/manual/peer2peer/js/main.js (hunk 6 above)
Comment (hkjellander) [inline src/manual/peer2peer/js/main.js]: Maybe this assumption should be clear (in a comment somewhere) to avoid confusion?

**Pull request merged**